Added PM2, tweaked readme to reflect PM2 and a little beautification

iamvikshan opened the pull request:
Removed forever manager and added PM2 manager instead. The file in question is [package.json](https://github.com/firtoz/GPT-Shell/blob/main/package.json). This meant changing start to `"start": "pm2 start ./lib/index.js --name GPT-Shell && pm2 save && pm2 logs"` and reflecting the changes in [README.md](https://github.com/firtoz/GPT-Shell/blob/main/README.md).

Misc changes: 
- Beautification mainly on links and commands to enable readers copy easily.
- Added example.env.local to ease setting up env vars and reflected it to README.md

firtoz commented:
Thank you for the pr.

Is there a particular reason to switch away from modern yarn?

iamvikshan commented:
Yw, 
What I did was remove Forever. Removing the Forever package and installing PM2 does not necessarily mean that I switched away from modern Yarn. Yarn is a package manager that can be used to manage dependencies for a Node.js project, and it is not limited to any specific package or tool.

Both Forever and PM2 are process managers for Node.js applications, and they can be used with any package manager, including Yarn. So, I just chose to continue using Yarn with PM2, as I 've demonstrated earlier by modifying the package.json file to use the yarn start command with PM2.

I guess your question is originating from the 300+ changed files, I was working from a codespace so it automatically updated yarn, so the only changes I made were on package.json and README.md, you can change those two files and close this pr, or lemme redo it prolly later when I'm on my local machine

Commit:
-forever, +pm2 => README.md

iamvikshan commented:
Alright, now I fixed the changes to only show on package.json and README.md

PS: Changed .gitignore to prevent pushing per machine configs, can leave as is or revert

iamvikshan committed:
Merge branch 'firtoz:main' into delta

firtoz commented:
Thank you, taking a look :)

firtoz commented on .gitignore:
```diff
@@ -14,3 +14,7 @@
 .env.local
 tmp
+node_modules
+.yarn/cache/*
```
We're using zero installs ( https://yarnpkg.com/features/zero-installs ) so don't need to ignore these:

`.yarn/cache/*`
`yarn.lock`

I think something in your configuration may have used `npm install` which doesn't work so well with yarn...

If it becomes a recurring issue we can make the project package manager agnostic.

iamvikshan replied:
I ignored `.yarn/cache/*` and `yarn.lock` because they were modified due to yarn updates, which i left out for you to review, and also after running `yarn remove forever`, it was the reason for 300+ file changes in my previous commits, so i chose to ignore them because once a user runs `yarn install`, `yarn.lock` is auto generated.

I installed pm2 using npm because yarn couldn't, i should have mentioned that, but pm2 installation is still covered in README.md, otherwise nothing else is `npm install`, just pm2

firtoz reviewed (approved):
Great work, thank you so much :D

I just need to consider moving away from zero-installs and modern yarn...

I guess it's very experimental still so ok, will clean up the modern yarn requirements before merging this PR

I need to finish up a few other things first but after those I'll come back to this :)

vikkshan commented:
Alr, honestly, I dodn't understand why you would need both npm and yarn; npm could be fine on its own, but I didn't want to move away from yarn because I haven't gone through the whole code

iamvikshan commented:
> Great work, thank you so much :D
> 
> I just need to consider moving away from zero-installs and modern yarn...
> 
> I guess it's very experimental still so ok, will clean up the modern yarn requirements before merging this PR
> 
> I need to finish up a few other things first but after those I'll come back to this :)

Alr, honestly, I dodn't understand why you would need both npm and yarn; npm could be fine on its own, but I didn't want to move away from yarn because I haven't gone through the whole code

firtoz commented:
Update: experimented a bit with not having the yarn cache etc and allowing typical `npm` and not "zero install" yarn, and it feels better, especially after testing with a few different hosting solutions.

So when I have free time hopefully in the next week or so, can do that and bring your PR in :)

Thanks!

firtoz commented:
Merged via #19 :) thanks!
